src/App.js

```diff
@@ -1,21 +1,92 @@
-import React, { Component } from "react";
-import { Route, BrowserRouter, Switch } from "react-router-dom";
-import DashboardPage from "./pages/DashboardPage";
-import SignIn from "./modules/sign/components/Signin";
+import React, { useReducer } from 'react';
+import { Route, BrowserRouter, Switch } from 'react-router-dom';
+import DashboardPage from './pages/DashboardPage';
+import SignIn from './modules/sign/components/Signin';
 
-class App extends Component {
-  render() {
-    return (
-      <BrowserRouter>
-          <Switch>
-            <Route exact path="/" component={DashboardPage} />
-            <Route path="/profile" component={DashboardPage} />
-            <Route path="/messages" component={DashboardPage} />
-            <Route path="/signin" component={SignIn} />
-          </Switch>
-      </BrowserRouter>
-    );
-  }
+export const UserContext = React.createContext();
```

Comment:
The context object, initialState and reducer could be moved to a separate file. Maybe a `store.js` file? The App.js file becomes unnecessarily long with this.